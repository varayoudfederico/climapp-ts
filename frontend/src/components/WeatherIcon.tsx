interface WeatherIconProps {
	iconText: string;
	size: string;
	color: string;
}

const WeatherIcon = ({ iconText, size, color }: WeatherIconProps) => {
	const renderIcon = () => {
		switch (iconText) {
			case "01n":
				return <i className="wi wi-night-clear wi-fw"></i>;
			case "02n":
				return <i className="wi wi-night-partly-cloudy wi-fw"></i>;
			case "03n":
				return <i className="wi wi-night-cloudy wi-fw"></i>;
			case "04n":
				return <i className="wi wi-night-cloudy wi-fw"></i>;
			case "09n":
				return <i className="wi wi-night-showers wi-fw"></i>;
			case "10n":
				return <i className="wi wi-night-rain wi-fw"></i>;
			case "11n":
				return <i className="wi wi-night-thunderstorm wi-fw"></i>;
			case "13n":
				return <i className="wi wi-night-snow wi-fw"></i>;
			case "50n":
				return <i className="wi wi-night-fog wi-fw"></i>;
			case "01d":
				return <i className="wi wi-day-sunny wi-fw"></i>;
			case "02d":
				return <i className="wi wi-day-cloudy wi-fw"></i>;
			case "03d":
				return <i className="wi wi-cloud wi-fw"></i>;
			case "04d":
				return <i className="wi wi-cloudy wi-fw"></i>;
			case "09d":
				return <i className="wi wi-day-showers wi-fw"></i>;
			case "10d":
				return <i className="wi wi-day-rain wi-fw"></i>;
			case "11d":
				return <i className="wi wi-day-thunderstorm wi-fw"></i>;
			case "13d":
				return <i className="wi wi-day-snow wi-fw"></i>;
			case "50d":
				return <i className="wi wi-day-fog wi-fw"></i>;
			case "senstermica":
				return <i className="wi wi-thermometer wi-fw"></i>;
			case "viento":
				return <i className="wi wi-strong-wind wi-fw"></i>;
			case "humedad":
				return <i className="wi wi-humidity wi-fw"></i>;
			case "presion":
				return <i className="wi wi-barometer wi-fw"></i>;
			case "amanecer":
				return <i className="wi wi-sunrise wi-fw"></i>;
			case "atardecer":
				return <i className="wi wi-sunset wi-fw"></i>;
			default:
				return <i className="wi wi-day-clear wi-fw"></i>;
		}
	};

	return <div style={{ color: color, fontSize: size }}>{renderIcon()}</div>;
};

export default WeatherIcon;
