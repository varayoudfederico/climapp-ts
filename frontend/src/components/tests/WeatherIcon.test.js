import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import WeatherIcon from "../WeatherIcon";

test("coincide con el snapshot de WeatherIcon", () => {
	const tree = renderer
		.create(
			<WeatherIcon iconText="01n" size="16px" color="#7754F8"></WeatherIcon>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
