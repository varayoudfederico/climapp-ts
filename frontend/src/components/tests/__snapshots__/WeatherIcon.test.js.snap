// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coincide con el snapshot de WeatherIcon 1`] = `
<div
  style={
    Object {
      "color": "#7754F8",
      "fontSize": "16px",
    }
  }
>
  <i
    className="wi wi-night-clear wi-fw"
  />
</div>
`;
